src/renderer/viz/expressions/buckets.js

```diff
@@ -139,29 +112,78 @@ export default class Buckets extends BaseExpression {
                 throw new Error(`buckets(): invalid ${getOrdinalFromIndex(index + 1)} parameter type\n\ttype was ${item.type}`);
             }
         });
+
+        this.numCategories = this.list.elems.length + 1;
+        this.numCategoriesWithoutOthers = this.input.type === 'category' ? this.numCategories - 1 : this.numCategories;
     }
 
     _applyToShaderSource (getGLSLforProperty) {
-        const childSources = this.childrenNames.map(name => this[name]._applyToShaderSource(getGLSLforProperty));
-        let childInlines = {};
-        childSources.map((source, index) => {
-            childInlines[this.childrenNames[index]] = source.inline;
+        const childSourcesArray = this.childrenNames.map(name => this[name]._applyToShaderSource(getGLSLforProperty));
+        let childSources = {};
+        childSourcesArray.map((source, index) => {
+            childSources[this.childrenNames[index]] = source;
         });
+
         const funcName = `buckets${this._uid}`;
         const cmp = this.input.type === 'category' ? '==' : '<';
+
+        // When there is "OTHERS" we don't need to take it into account
+        const divisor = this.numCategoriesWithoutOthers - 1 || 1;
+
         const elif = (_, index) =>
-            `${index > 0 ? 'else' : ''} if (x${cmp}(${childInlines[`arg${index}`]})){
-                return ${index}.;
+            `${index > 0 ? 'else' : ''} if (x${cmp}(${childSources.list.inline[index]})){
+                return ${index}./${divisor.toFixed(20)};
             }`;
         const funcBody = this.list.elems.map(elif).join('');
         const preface = `float ${funcName}(float x){
             ${funcBody}
-            return ${this.numCategories - 1}.;
+            return ${this.input.type === 'category' ? OTHERS_GLSL_VALUE : (this.numCategories - 1).toFixed(20)};
         }`;
 
         return {
-            preface: this._prefaceCode(childSources.map(s => s.preface).reduce((a, b) => a + b, '') + preface),
-            inline: `${funcName}(${childInlines.input})`
+            preface: this._prefaceCode(childSourcesArray.map(s => s.preface).join('\n') + preface),
```

Comment:
I think `childSourcesArray.map(s => s.preface).join('\n')` can be replaced by `childSources.list.preface`